cpmpy/expressions/utils.py

```diff
@@ -76,7 +79,11 @@ def argval(a):
         
         We check with hasattr instead of isinstance to avoid circular dependency
     """
-    return a.value() if hasattr(a, "value") else a
+    try:
+        return a.value() if hasattr(a, "value") else a
+    except IncompleteFunctionError as e:
+        if a.is_bool(): return False
```

Comment:
One problem that could arise is: `(x/0>0) == (x/0<0)` which seems a bit counter-intuitive.